addons/binding/org.openhab.binding.sleepiq/src/main/java/org/openhab/binding/sleepiq/handler/SleepIQDualBedHandler.java

```diff
@@ -0,0 +1,135 @@
+        }
+
+        ThingHandler handler = bridge.getHandler();
+        if (!(handler instanceof SleepIQCloudHandler)) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "Incorrect bridge thing found");
+            return;
+        }
+
+        SleepIQCloudHandler cloudHandler = (SleepIQCloudHandler) handler;
+
+        logger.debug("Reading SleepIQ bed binding configuration");
+        bedId = getConfigAs(SleepIQBedConfiguration.class).bedId;
+
+        logger.debug("Updating SleepIQ bed properties for bed {}", bedId);
+        updateProperties(cloudHandler.updateProperties(cloudHandler.getBed(bedId), editProperties()));
+        updateStatus(ThingStatus.ONLINE);
+
+        logger.debug("Registering SleepIQ bed status listener");
+        cloudHandler.registerBedStatusListener(this);
+    }
+
+    @Override
+    public void dispose() {
+
+        logger.debug("Disposing bed handler for bed {}", bedId);
+
+        Bridge bridge = getBridge();
+        if (bridge != null) {
+            BridgeHandler bridgeHandler = bridge.getHandler();
+            if (bridgeHandler instanceof SleepIQCloudHandler) {
+                ((SleepIQCloudHandler) bridgeHandler).unregisterBedStatusListener(this);
+            }
+        }
+
+        bedId = null;
+    }
+
+    @Override
+    public void onBedStateChanged(final SleepIQ cloud, final BedStatus status) {
+
+        if (!status.getBedId().equals(bedId)) {
+            return;
+        }
+
+        logger.debug("Updating left side status for bed {}", bedId);
+        BedSideStatus left = status.getLeftSide();
+        updateState(CHANNEL_LEFT_IN_BED, left.isInBed() ? OnOffType.ON : OnOffType.OFF);
+        updateState(CHANNEL_LEFT_SLEEP_NUMBER, new DecimalType(left.getSleepNumber()));
+        updateState(CHANNEL_LEFT_PRESSURE, new DecimalType(left.getPressure()));
+        updateState(CHANNEL_LEFT_LAST_LINK, new StringType(left.getLastLink().toString()));
+        updateState(CHANNEL_LEFT_ALERT_ID, new DecimalType(left.getAlertId()));
+        updateState(CHANNEL_LEFT_ALERT_DETAILED_MESSAGE, new StringType(left.getAlertDetailedMessage()));
+
+        logger.debug("Updating right side status for bed {}", bedId);
+        BedSideStatus right = status.getRightSide();
+        updateState(CHANNEL_RIGHT_IN_BED, right.isInBed() ? OnOffType.ON : OnOffType.OFF);
+        updateState(CHANNEL_RIGHT_SLEEP_NUMBER, new DecimalType(right.getSleepNumber()));
+        updateState(CHANNEL_RIGHT_PRESSURE, new DecimalType(right.getPressure()));
+        updateState(CHANNEL_RIGHT_LAST_LINK, new StringType(right.getLastLink().toString()));
+        updateState(CHANNEL_RIGHT_ALERT_ID, new DecimalType(right.getAlertId()));
+        updateState(CHANNEL_RIGHT_ALERT_DETAILED_MESSAGE, new StringType(right.getAlertDetailedMessage()));
+    }
+
+    @Override
+    public void handleCommand(final ChannelUID channelUID, final Command command) {
+        // all channels are read-only
```

Comment:
@kaikreuzer 

I ran into a small issue with Guava. According to the openhab-debs-repo the project is using Guava version 10.0.1.v201203051515, but when I deploy my binding to my running OH 2.1.0-SNAPSHOT instance, I'm getting NoSuchMethodError. When I check the bundle:list on that instance, I'm seeing this:

```
15 | Active   |  80 | 18.0.0                | Guava: Google Core Libraries for Java
```

By version 18, the CacheBuilder API has changed slightly. I haven't been able to make version 18 load in the Eclipse project though.

What is the official version of Guava that should be available in the OH runtime? If it is now 18, what setting or config do I need to update to get back in line with current?